aws-lambda/src/databricks_cdk/resources/permissions/changes.py

```diff
@@ -0,0 +1,87 @@
+from databricks.sdk.service.catalog import PermissionsChange, PermissionsList, Privilege, PrivilegeAssignment
+
+
+def get_assignment_dict_from_permissions_list(permissions_list: PermissionsList) -> dict[str, list[Privilege]]:
+    """Converts PermissionsList to dict with key of principal and list of associated privileges as value"""
+    privilige_assignments = permissions_list.privilege_assignments or {}
+
+    if privilige_assignments is None:
+        return {}
+
+    return {
+        x.principal: [Privilege(y) for y in x.privileges]
+        for x in privilige_assignments
+        if permissions_list.privilege_assignments is not None and x.principal is not None and x.privileges is not None
```

Comment:
privilege_assignments can't be None here

Reply:
yes good spot. Removed condition